cyberdrop_dl/managers/download_manager.py

```diff
@@ -17,23 +19,80 @@
 
 
 class FileLocksVault:
-    """Is this necessary? No. But I want it."""
+    """File locks vault with automatic cleanup of unused locks to prevent memory leaks."""
 
     def __init__(self) -> None:
         self._locked_files: dict[str, asyncio.Lock] = {}
```

Comment:
Thanks. This is definitely needed. However, we can get all of this for free by using a weakref dict.

This will be the only change needed:

```suggestion
        self._locked_files: weakref.WeakValueDictionary[str, asyncio.Lock] = weakref.WeakValueDictionary()
```